src/rules/double-slash-comment-empty-line-before/index.js

```diff
@@ -90,6 +90,28 @@ export default function(expectation, options) {
         return;
       }
 
+      if (context.fix) {
+        const fix = (comment, match, replace) => {
```

Comment:
This much cleaner than my PR!

One thing caught my eye though. The fix function is now inside the `root.walkComments` loop.  I think that it would be nicer to move that function outside the loop so that it's not re-declared on every loop:

```js
    const fix = (comment, match, replace) => {
      const escapedMatch = match.replace(
        /(\r)?\n/g,
        (_, r) => (r ? "\\r\\n" : "\\n")
      );
      comment.raws.before = comment.raws.before.replace(
        new RegExp(`^${escapedMatch}`),
        replace
      );
    };

    root.walkComments(comment => {
```